Getting Started documentation update.

ewaterlander opened the pull request:
* Put Core Build and Managed Build items in separate topics.
* Added two images to Creating a CMake project.
* Added new pages about creating a CBS Makefile project, using existing code, and the launch bar.
* Renamed "Creating a simple application" to "Creating a Managed Build System Makefile project" and replaced two images.
* Removed "Creating a Makefile project", because it was duplicate information.

Fixes #992

ewaterlander commented:
Hi @jonahgraham @jld01 @betamaxbandit ,

Would you like to review this change? It is not finished yet. I'm going to add topics about Building, Running, and Debugging. You can start adding comments.

I separated CBS and MBS.

![image](https://github.com/user-attachments/assets/8014b748-bb2a-46e3-8fff-2c3451d85014)

jonahgraham commented:
I'll have a look soon - in the meantime since I don't have an auto comment, the docs are available from this build in https://github.com/eclipse-cdt/cdt/actions/runs/12933695063?pr=1056 (docs link near the bottom)

jonahgraham commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_launchbar.adoc:
```diff
@@ -0,0 +1,52 @@
+image:launchbar.png[Launch Bar]
+
+. Build button, to launch a build.
+. Run button, to launch a run. This button will change to a Debug
```
Should we mention profile mode somewhere here too?

ewaterlander replied:
I was doubting about mentioning it in "getting started". The Profile mode is not in the launch mode selector before you run profiling. Profiling only works properly in Makefile projects, not in CMake. And you need to have the profiling plugin installed before it is available at all. I think it is more an advanced topic.

jonahgraham reviewed (approved):
Overall this looks very good, I think how you have split it up is very useful.

jld01 commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_launchbar.adoc:
```diff
@@ -0,0 +1,52 @@
+
+CBS projects rely on the *LaunchBar*. Make sure the launch bar plugin
+is installed and enabled. In the global preferences the launch bar can
+be enabled or disabled.
```
```suggestion
be enabled or disabled:
```

jld01 commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_launchbar.adoc:
```diff
@@ -0,0 +1,52 @@
+
+== Launch Bar
+
+CBS projects rely on the *LaunchBar*. Make sure the launch bar plugin
```
```suggestion
Core Build System (CBS) projects rely on the *Launch Bar*. Make sure the _LaunchBar UI_ feature
```

jld01 commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_using_existing_code.adoc:
```diff
@@ -0,0 +1,66 @@
+
+== Using existing code
+
+This tutorial describes how to use existing code for Core Build System
+projects.
```
```suggestion
This tutorial describes how to create a new Core Build System
project for use with existing code.
```

jld01 commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_using_existing_code.adoc:
```diff
@@ -0,0 +1,66 @@
+Using existing code is done by creating a new **empty** project on top of the
+existing code.
+
+=== Using an existing CMake project
+
+To create an empty CMake project select the template *Empty or Existing CMake
+Project* in the *New C/{cpp} Project* wizard.
```
Need to clarify the distinction between a "CMake project" (IDE independent) and a "CMake project" (metadata created by CDT).

jld01 commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_using_existing_code.adoc:
```diff
@@ -0,0 +1,66 @@
+More information about creating a CMake project is described in
+xref:new_cmake_proj.adoc[Creating a CMake project]
+
+=== Using an existing Makefile project
+
+Create a Makefile project as described in
```
Similarly, distinguish the two uses of "Makefile project" here.

jld01 commented on doc/org.eclipse.cdt.doc.user/src/getting_started/new_cbs_makefile_proj.adoc:
```diff
@@ -0,0 +1,78 @@
+To create a project:
+
+. Select menu:File[New > Project].
+
+
```
I think multiple empty lines are being interpreted as the end of the numbered list. So the next numbered list item will be item "1" again. Try a single empty line only.

jonahgraham replied:
I had some problems with keeping numbered lists together too, I ended up having to add `+` at the beginning of the blank lines to show list continuation. See https://docs.asciidoctor.org/asciidoc/latest/lists/continuation/#list-continuation

jld01 reviewed (commented):
A few comments on your work-in-progress, @ewaterlander. Great to see the documentation being updated.

ewaterlander commented:
I added three new topics: Building, Running, and Debugging a project. Now the change is complete. I will process the comments on Monday. Thanks. Have a nice weekend!

ewaterlander commented:
Hi,
I processed the comments.

betamaxbandit commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_using_existing_code.adoc:
```diff
@@ -0,0 +1,66 @@
+== Using existing code
+
+This tutorial describes how to create a new Core Build System
+(CBS) project for use with existing code.
```
It assumes existing code is located in a directory containing source files. Note, CDT project files are written into this directory.

ewaterlander replied:
Yes

betamaxbandit commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_using_existing_code.adoc:
```diff
@@ -0,0 +1,66 @@
+This tutorial describes how to create a new Core Build System
+(CBS) project for use with existing code.
+
+Using existing code is done by creating a new **empty** project on top of the
+existing code.
```
[I think this task is quite unintuitive for the new user, so it's worth labouring the point about project files, directories etc]
Using existing code is performed by creating a new **empty** CDT project in the top directory containing the existing source files. With this method, new project files are created, but no new project directory is created.

ewaterlander replied:
Done

betamaxbandit commented on doc/org.eclipse.cdt.doc.user/src/getting_started/cbs_using_existing_code.adoc:
```diff
@@ -0,0 +1,66 @@
+
+image:cbs_empty_cmake_project.png[Empty or Existing CMake Project]
+
+In the *New CMake Project* wizard deselect *Use default location*
```
[Consider replacing "deselect" with "uncheck". Feels a little more user friendly. There are a couple more occurrences of this below]

ewaterlander replied:
Done

betamaxbandit reviewed (commented):
Hi @ewaterlander ,
thanks for updating the docs.
I've made a few suggestions.
Cheers John

jonahgraham reviewed (approved):
I think @betamaxbandit's comments are worth addressing now, or in a follow-up. I have a TODO in the CDT 12 N&N to update the screenshot:

https://github.com/eclipse-cdt/cdt/blob/84d99f277ab1e48da8e1c8ce3e6c15d1e8b260ad/NewAndNoteworthy/CDT-12.0.md#L23

But there is some flexibility on that file being updated because it can happen after our last build. The screenshots in the online help need to be updated before final build to populate https://help.eclipse.org/latest/index.jsp and built-in help

I have created #1064 and added updating doc/org.eclipse.cdt.doc.user/images/cbs_build_settings_tab_cmake.png to the check-list.

ewaterlander commented:
Hi,
I will address the new comments tomorrow. I don't mind doing it in this PR or a follow up PR.

jld01 commented:
> Could you have a look again? I think the work is done now.

Thanks for the updates, @ewaterlander.

ewaterlander commented:
Hi @jonahgraham and @betamaxbandit ,

I updated the image and addressed the comments.

ewaterlander commented:
I squashed and rebased the commit. After rebasing I lost some time, because Eclipse-Committers did not properly refresh the work area, which cause build errors in my environment. :(

jonahgraham commented:
> I squashed and rebased the commit. After rebasing I lost some time, because Eclipse-Committers did not properly refresh the work area, which cause build errors in my environment. :(

That is frustrating - I think the other day I had a fail to refresh properly error, I couldn't reproduce it and I am not sure what happened.

jonahgraham commented:
Thanks @ewaterlander
